src/AutoRest.CSharp/Common/AutoRest/Plugins/NewProjectScaffolding.cs

```diff
@@ -0,0 +1,429 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Json;
+using System.Text;
+using System.Threading.Tasks;
+using AutoRest.CSharp.AutoRest.Communication;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Models.Types;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace AutoRest.CSharp.Common.AutoRest.Plugins
+{
+    internal class NewProjectScaffolding
+    {
+        private string _serivceDirectoryName;
+        private string _projectDirectory;
+        private string _testDirectory;
+        private string _serviceDirectory;
+        private bool _isAzureSdk;
+
+        public NewProjectScaffolding()
+        {
+            _serivceDirectoryName = Path.GetFileName(Path.GetFullPath(Path.Combine(Configuration.AbsoluteProjectFolder, "..", "..")));
+            _projectDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..");
+            _testDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..", "tests");
+            _serviceDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..", "..");
+            _isAzureSdk = Configuration.Namespace.StartsWith("Azure.");
+        }
+
+        public async Task<bool> Execute()
+        {
+            if (!_isAzureSdk)
+            {
+                //clean up old sln and csproj files
+                foreach (var file in Directory.GetFiles(Configuration.AbsoluteProjectFolder, "*.csproj"))
```

Comment:
Couldn't we do this in one pass if we tell `GetFiles` to include subdirectories and expand the filter to both clauses?